Stabilize selenium tests.

T4rk1n opened the pull request:
- Runs threaded instead of multiprocess, (faster, more stable).
- Login container standardized to `react-entry-point` (same as dash)
- Wait for `react-entry-point` after navigating to the page.

scjody commented on tests/IntegrationTests.py:
```diff
@@ -10,38 +15,25 @@
-            time.sleep(0.25)
-        if latest_exception:
-            raise latest_exception
+        return WebDriverWait(self.driver, 10).until(
```
Neat, I didn't know about `WebDriverWait`. Don't you want to use `TIMEOUT` instead of `10` though?

scjody commented on tests/IntegrationTests.py:
```diff
@@ -10,38 +15,25 @@
-            time.sleep(0.25)
-        if latest_exception:
-            raise latest_exception
+        return WebDriverWait(self.driver, 10).until(
```
`TIMEOUT`?

scjody commented on tests/IntegrationTests.py:
```diff
@@ -59,9 +51,10 @@ def setUp(self):
-        self.server_process.terminate()
-        time.sleep(5)
+        time.sleep(2)
+        # self.server_process.terminate()
```
Can you remove this rather than commenting it out? We have the git history if we ever want to go back to using a server process.

scjody commented on tests/IntegrationTests.py:
```diff
@@ -70,20 +63,30 @@ def run():
+
+        self.server_thread = threading.Thread(target=run)
+        self.server_thread.start()
+        # self.server_process = multiprocessing.Process(target=run)
```
This too :)

T4rk1n commented:
Those changes stabilized a bit the tests but there's still some times it fails, last fail chrome crashed. I'm gonna try to remove tox and see if it improves.

T4rk1n commented:
@scjody

I removed anything that could induce instability from the code.

- I think running threaded is more stable since spawning threads is faster so waiting for the server to start don't fail. It also runs 2 minutes faster and can run on windows.
- Removed tox, we don't use it anymore in the other repos.
- Runs the unittests directly with the unittests module.
- There was two selenium instance, one as class instance and the other as `self`. I thought it was clashing, but only keeping one did not improve stability (but improved run time a bit).
- Changed manual text asserts to wait_for_text, improved a bit.
- All selenium elements find should use wait_for_element_by_css_selector so it can wait for the timeout before failing.
- Timeout should be > 20 to consistently succeeds.
- Chrome sometimes crash randomly when the tests starts, did not find a cause.

The tests are still flaky, I think there's a throttle or something on some of the requests we make to streambed ?
- Seems particularly bad with the logout test and subsequent tests fail after timing out on elements.
- The first container to start the tests seems to succeeds while the second fail more often than not.

Also this is merging in your kerberos branch because I made a mistake and started the branch on it. I can make a patch and a new PR if it's better.

T4rk1n commented:
I think if the requests status was 429, it would be raised when the requests are made thru api_requests so they would show as the cause of failure, but they are not. I think some of the auth requests come from the popup, need to find a way to get the logs of those.

I don't know about the user, I think you need to be a paid user to create dash auth apps ?

T4rk1n commented:
Closing this in favour of #68, had trouble changing the root of the branch and it was easier to make a patch and a new PR.
